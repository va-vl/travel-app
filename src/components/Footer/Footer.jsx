import * as React from 'react';
import styles from './styles/styles';

const urlRss = 'https://rs.school/js/';
const urlGithubPrefix = 'https://github.com/';

const Footer = () => {
  const classes = styles();
  const developers = [
    'velidoss',
    'arumirinka',
    'va-z',
    'reagentjs',
  ];
  const developersJsx = developers.map((dev) => (
    <a
      key={dev}
      href={urlGithubPrefix + dev}
      target="_blank"
      rel="noreferrer"
      className="github-logo"
      data-title={dev}
    >
      <img
        src="/icons/github.svg"
        alt="github-logo"
        width="30"
      />
    </a>
  ));

  return (
    <footer className={classes.root}>
      <div>
        <a
          href={urlRss}
          target="_blank"
          rel="noreferrer"
          className="rss-logo"
        >
          <img
            src="/icons/rolling-scopes.svg"
            alt="RSSchool-logo"
            width="75"
          />
        </a>
        <h4>2021</h4>
      </div>
      <div>
        {developersJsx}
      </div>
    </footer>
  );
};

export default Footer;
