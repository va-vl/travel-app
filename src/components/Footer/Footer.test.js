import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Footer from './Footer';

test('component Footer renders', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer></Footer>, div);
});

test('render correctly Footer component', () => {
  const FooterComponent = TestRenderer.create(<Footer />).toJSON();
  expect(FooterComponent).toMatchSnapshot();
});
