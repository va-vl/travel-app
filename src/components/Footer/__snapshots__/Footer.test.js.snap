// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Footer component 1`] = `
<footer
  className="makeStyles-root-1"
>
  <div>
    <a
      className="rss-logo"
      href="https://rs.school/js/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="RSSchool-logo"
        src="logo-rolling-scopes.svg"
        width="75"
      />
    </a>
    <h4>
      2021
    </h4>
  </div>
  <div>
    <a
      className="github-logo"
      data-title="velidoss"
      href="https://github.com/velidoss"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="github-logo"
        src="logo-github.svg"
        width="30"
      />
    </a>
    <a
      className="github-logo"
      data-title="arumirinka"
      href="https://github.com/arumirinka"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="github-logo"
        src="logo-github.svg"
        width="30"
      />
    </a>
    <a
      className="github-logo"
      data-title="va-z"
      href="https://github.com/va-z"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="github-logo"
        src="logo-github.svg"
        width="30"
      />
    </a>
    <a
      className="github-logo"
      data-title="reagentjs"
      href="https://github.com/reagentjs"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="github-logo"
        src="logo-github.svg"
        width="30"
      />
    </a>
  </div>
</footer>
`;
